server/cache.go

```diff
@@ -374,28 +388,50 @@ func (r *regionsInfo) randRegion(storeID uint64) *metapb.Region {
 	return nil
 }
 
+type storeStatus struct {
+	// store capacity info.
+	stats *pdpb.StoreStats
+
+	leaderPeerCount int
+}
+
+func (s *storeStatus) clone() *storeStatus {
+	return &storeStatus{
+		stats:           proto.Clone(s.stats).(*pdpb.StoreStats),
+		leaderPeerCount: s.leaderPeerCount,
+	}
+}
+
 // storeInfo is store cache info.
 type storeInfo struct {
 	store *metapb.Store
 
-	// store capacity info.
-	stats *pdpb.StoreStats
+	stats *storeStatus
 }
 
 func (s *storeInfo) clone() *storeInfo {
 	return &storeInfo{
 		store: proto.Clone(s.store).(*metapb.Store),
-		stats: proto.Clone(s.stats).(*pdpb.StoreStats),
+		stats: s.stats.clone(),
 	}
 }
 
 // usedRatio is the used capacity ratio of storage capacity.
 func (s *storeInfo) usedRatio() float64 {
-	if s.stats.GetCapacity() == 0 {
+	if s.stats.stats.GetCapacity() == 0 {
+		return 0
+	}
+
+	return float64(s.stats.stats.GetCapacity()-s.stats.stats.GetAvailable()) / float64(s.stats.stats.GetCapacity())
+}
+
+// leaderScore is the leader peer count score of store.
+func (s *storeInfo) leaderScore(regionCount int) float64 {
```

Comment:
`regionCount` is only the region count in this store, not the total region count.